src/zenml/zen_service/zen_service_api.py

```diff
@@ -58,9 +77,34 @@
         f"{ENV_ZENML_PROFILE_NAME} to specify the use of a profile "
         "other than the currently active one)"
     )
-stack_store: BaseStackStore = Repository.create_store(profile)
+zen_store: BaseZenStore = Repository.create_store(
+    profile, skip_default_registrations=True
+)
+
+
+security = HTTPBasic()
+
+
+def authorize(credentials: HTTPBasicCredentials = Depends(security)) -> None:
+    """Authorizes any request to the ZenService.
+
+    Right now this method only checks if the username provided as part of http
+    basic auth credentials is registered in the ZenStore.
+
+    Args:
+        credentials: HTTP basic auth credentials passed to the request.
+    """
+    try:
+        zen_store.get_user(credentials.username)
+    except KeyError:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Incorrect username.",
+        )
 
-app = FastAPI(title="ZenML", version="0.7.0")
+
+app = FastAPI(title="ZenML", version=zenml.__version__)
+requires_authorization = APIRouter(dependencies=[Depends(authorize)])
```

Comment:
Yep right at the bottom, and unfortunately needs to be there as well as all the get/post/delete methods need to be declared before it is added to the `app`. Does vim not have autocomplete or why do we need shorter names? :P
Any suggestions, the only thing that I can think of would be `authorized`?

Reply:
ctrl-N for autocomplete in vim :D

I just mean because it turns a bunch of the decorators from 1-liners into 3 lines (thx to 80 char line width), and this is already such a long file. I was going to say `authed` but I guess yours would be more grammatically corect